gazebo_ros/src/node.cpp

```diff
@@ -98,13 +103,35 @@ Node::SharedPtr Node::Get(sdf::ElementPtr sdf)
     }
   }
 
+  // set full node name
+  if (ns[0] == '/' && ns.size() == 1) {
+    full_name = ns + name;
+  } else {
+    full_name = ns + "/" + name;
+  }
+
+  // check if node with the same name exists already
+  if (static_existing_nodes_.check_node(full_name)) {
+    RCLCPP_ERROR(
+      internal_logger(),
+      "Found multiple nodes with same name: %s. This might be due to multiple plugins using same "
+      "name to solve this either change one of the the plugin names or use a different namespace. "
+      "The error might also result from a custom plugin inheriting from one of the GazeboRosPlugin "
+      "this can be solved by accessing node object of the parent class itself instead of creating "
+      "a new node object for custom plugin.", full_name.c_str());
```

Comment:
I am moving forward with having an optional argument `Get(sdf::ElementPtr sdf, std::string node_name = "")` which will override the plugin name in case the default node name is provided.